internal/corazawaf/body_buffer.go

```diff
@@ -61,6 +62,10 @@ func (br *BodyBuffer) Write(data []byte) (n int, err error) {
 		targetLen = br.length + int64(len(data))
 	}
 
+	if targetLen >= br.options.Limit {
```

Comment:
I would say this check goes first, right before the emptiness of the data. Something like

```
if br.length == br.options.Limit || br.length >= (br.options.Limit -int64(len(data))) {
    return 0, errors.New("Limit reached while writing")
}
```

I would say, beyond this point you don't need to check the overflow because if `br.options.Limit == math.MaxIn64` you are safe and if `len(data) == math.MaxIn64` still you don't get the overflow. What do you think?

Reply:
That makes sense, thanks JC!